q2_vizard/tests/test_lineplot.py

```diff
@@ -0,0 +1,192 @@
+                ValueError,
+                'Replicates found in `x`.'
+            ):
+                lineplot(output_dir=output_dir, metadata=self.md,
+                         x_measure='x', y_measure='y')
+
+    # selenium testing
+    def _selenium_lineplot_test(self, driver, x_measure, y_measure,
+                                facet_measure, replicate_method, exp_subtitle,
+                                exp_legend, exp_marks_len, exp_mark_class,
+                                exp_x_mark, exp_y_mark):
+        with tempfile.TemporaryDirectory() as output_dir:
+            lineplot(
+                output_dir=output_dir, metadata=self.md,
+                x_measure=x_measure, y_measure=y_measure,
+                facet_by=facet_measure,
+                replicate_method=replicate_method
+            )
+
+            driver.get(f"file://{os.path.join(output_dir, 'index.html')}")
+
+            # test that our axes match xy input measures
+            axis_elements = \
+                driver.find_elements(By.CSS_SELECTOR, 'g.mark-group.role-axis')
+            self.assertEqual(len(axis_elements), 2)
+
+            for _, axis in enumerate(axis_elements):
+                label = axis.get_attribute('aria-label')
+                if 'X-axis' in label:
+                    self.assertIn(f"axis titled '{x_measure}'", label)
+                elif 'Y-axis' in label:
+                    self.assertIn(f"axis titled '{y_measure}'", label)
+                else:
+                    raise ValueError(f'Unexpected axis element {label} found.')
+
+            # test that the legend contains the correct group
+            legend_element = \
+                driver.find_element(By.CSS_SELECTOR,
+                                    'g.mark-group.role-legend')
+
+            label = legend_element.get_attribute('aria-label')
+            self.assertIn(exp_legend, label)
+
+            # test that the title class contains the correct subtitle text
+            title_element = \
+                driver.find_element(By.CSS_SELECTOR, 'g.mark-group.role-title')
+
+            subtitle = title_element.get_attribute('subtitle')
+            self.assertEqual(subtitle, exp_subtitle)
+
+            # test that the correct number of line marks are present
+            # equal to the number of unique groups in the facet_by, else = 1
+            line_elements = \
+                driver.find_elements(By.CSS_SELECTOR, 'g.mark-line.role-mark')
+            md_df = self.md.to_dataframe()
+
+            if facet_measure:
+                exp_lines = len(md_df[facet_measure].unique())
+            else:
+                exp_lines = 1
+
+            self.assertEqual(len(line_elements), exp_lines)
+
+            # test that the correct number of scatter marks are present
+            # and a mark is where we expect it to be
+            # for some reason this viz renders multiple groups
+            # of the same marks, so just selecting the first one
```

Comment:
Ah, that's because the spec actually encodes the marks from the global `scatter_table` in each line facet. You could move those to a global mark, since they don't depend on the `line_table` group and then still color by the color_scale and field.